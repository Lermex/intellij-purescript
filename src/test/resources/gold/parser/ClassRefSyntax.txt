Purescript File
  PSProgramImpl(Program)
    PSModuleImpl(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSModuleNameImpl(ModuleName)
        PSQualifiedImpl(Qualified)
          PsiElement(proper name)('Main')
        PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSNamedBinderImpl(NamedBinder)
        <empty list>
      PSImportDeclarationImpl(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSImportModuleNameImpl(ImportModuleName)
          PSQualifiedImpl(Qualified)
            PsiElement(proper name)('Lib')
          PsiElement(whitespace)(' ')
        PsiElement(()('(')
        PSPositionedDeclarationRefImpl(PositionedDeclarationRef)
          PsiElement(class)('class')
          PsiElement(whitespace)(' ')
          PSClassNameImpl(ClassName)
            PSQualifiedImpl(Qualified)
              PSProperNameImpl(ProperName)
                PsiElement(proper name)('X')
        PsiElement(,)(',')
        PsiElement(whitespace)(' ')
        PSPositionedDeclarationRefImpl(PositionedDeclarationRef)
          PSValueRefImpl(ValueRef)
            PSIdentifierImpl(identifier)
              PsiElement(identifier)('go')
        PsiElement())(')')
        PsiElement(whitespace)('\n')
      PSNamedBinderImpl(NamedBinder)
        <empty list>
      PSImportDeclarationImpl(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSImportModuleNameImpl(ImportModuleName)
          PSQualifiedImpl(Qualified)
            PsiElement(proper name)('Control')
            PsiElement(.)('.')
            PsiElement(proper name)('Monad')
            PsiElement(.)('.')
            PsiElement(proper name)('Eff')
            PsiElement(.)('.')
            PsiElement(proper name)('Console')
          PsiElement(whitespace)(' ')
        PsiElement(()('(')
        PSPositionedDeclarationRefImpl(PositionedDeclarationRef)
          PSValueRefImpl(ValueRef)
            PSIdentifierImpl(identifier)
              PsiElement(identifier)('log')
        PsiElement())(')')
        PsiElement(whitespace)('\n\n')
      PSTypeDeclarationImpl(TypeDeclaration)
        PSTypeAnnotationNameImpl(TypeAnnotationName)
          PSIdentifierImpl(identifier)
            PsiElement(identifier)('go'')
            PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSTypeImpl(Type)
          PSTypeAtomImpl(TypeAtom)
            PSForAllImpl(ForAll)
              PsiElement(forall)('forall')
              PsiElement(whitespace)(' ')
              PSIdentifierImpl(GenericIdentifier)
                PsiElement(identifier)('a')
              PsiElement(.)('.')
              PsiElement(whitespace)(' ')
              PSConstrainedTypeImpl(ConstrainedType)
                PSTypeImpl(Type)
                  PSTypeAtomImpl(TypeAtom)
                    PSTypeConstructorImpl(TypeConstructor)
                      PSQualifiedImpl(Qualified)
                        PSProperNameImpl(ProperName)
                          PsiElement(proper name)('X')
                          PsiElement(whitespace)(' ')
                  PSTypeAtomImpl(TypeAtom)
                    PSIdentifierImpl(GenericIdentifier)
                      PsiElement(identifier)('a')
                      PsiElement(whitespace)(' ')
                  PsiElement(=>)('=>')
                  PsiElement(whitespace)(' ')
                  PSTypeImpl(Type)
                    PSTypeAtomImpl(TypeAtom)
                      PSIdentifierImpl(GenericIdentifier)
                        PsiElement(identifier)('a')
                        PsiElement(whitespace)(' ')
                    PsiElement(->)('->')
                    PsiElement(whitespace)(' ')
                    PSTypeImpl(Type)
                      PSTypeAtomImpl(TypeAtom)
                        PSIdentifierImpl(GenericIdentifier)
                          PsiElement(identifier)('a')
                          PsiElement(whitespace)('\n')
      PSValueDeclarationImpl(ValueDeclaration)
        PSIdentifierImpl(identifier)
          PsiElement(identifier)('go'')
          PsiElement(whitespace)(' ')
        PSNamedBinderImpl(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSValueImpl(Value)
          PSPrefixValueImpl(PrefixValue)
            PSVarImpl(Var)
              PSQualifiedImpl(Qualified)
                PSIdentifierImpl(identifier)
                  PsiElement(identifier)('go')
                  PsiElement(whitespace)('\n\n')
      PSValueDeclarationImpl(ValueDeclaration)
        PSIdentifierImpl(identifier)
          PsiElement(identifier)('main')
          PsiElement(whitespace)(' ')
        PSNamedBinderImpl(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSValueImpl(Value)
          PSPrefixValueImpl(PrefixValue)
            PSVarImpl(Var)
              PSQualifiedImpl(Qualified)
                PSIdentifierImpl(identifier)
                  PsiElement(identifier)('log')
                  PsiElement(whitespace)(' ')
            PSStringLiteralImpl(StringLiteral)
              PsiElement(string)('"Done"')