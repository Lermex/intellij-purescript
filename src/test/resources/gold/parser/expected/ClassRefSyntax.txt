Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(Qualified)
        PSASTWrapperElement(ModuleName)
          PsiElement(proper name)('Main')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Lib')
        PsiElement(whitespace)(' ')
        PsiElement(()('(')
        PSASTWrapperElement(PositionedDeclarationRef)
          PsiElement(class)('class')
          PsiElement(whitespace)(' ')
          PsiElement(proper name)('X')
        PsiElement(,)(',')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(PositionedDeclarationRef)
          PSASTWrapperElement(ValueRef)
            PsiElement(identifier)('go')
        PsiElement())(')')
        PsiElement(whitespace)('\n')
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Control')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Monad')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Eff')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Console')
        PsiElement(whitespace)(' ')
        PsiElement(()('(')
        PSASTWrapperElement(PositionedDeclarationRef)
          PSASTWrapperElement(ValueRef)
            PsiElement(identifier)('log')
        PsiElement())(')')
        PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeDeclaration)
        PsiElement(identifier)('go'')
        PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PSASTWrapperElement(ForAll)
              PsiElement(forall)('forall')
              PsiElement(whitespace)(' ')
              PsiElement(identifier)('a')
              PsiElement(.)('.')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(ConstrainedType)
                PSASTWrapperElement(Type)
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeConstructor)
                      PSASTWrapperElement(Qualified)
                        PsiElement(proper name)('X')
                        PsiElement(whitespace)(' ')
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeVar)
                      PsiElement(identifier)('a')
                      PsiElement(whitespace)(' ')
                  PsiElement(=>)('=>')
                  PsiElement(whitespace)(' ')
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(TypeVar)
                        PsiElement(identifier)('a')
                        PsiElement(whitespace)(' ')
                    PsiElement(->)('->')
                    PsiElement(whitespace)(' ')
                    PSASTWrapperElement(Type)
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(TypeVar)
                          PsiElement(identifier)('a')
                          PsiElement(whitespace)('\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('go'')
        PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('go')
                PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('main')
        PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('log')
                PsiElement(whitespace)(' ')
            PSASTWrapperElement(StringLiteral)
              PsiElement(string)('"Done"')